src/client.py

```diff
@@ -1,32 +1,40 @@
+# coding=utf-8
 """Something client-y."""
 
 import sys
 import socket
 
 
 def client(message):
-
-    message += 'EOF'
+    """Client side of an echo server."""
+    if sys.version_info[0] == 3:
+        try:
+            message = message.decode('utf8')
+        except AttributeError:
+            pass
+    message = str(len(message)) + '.' + message
     print('message to send is: ' + message)
     infos = socket.getaddrinfo('127.0.0.1', 5001)
     stream_info = [i for i in infos if i[1] == socket.SOCK_STREAM][0]
     client = socket.socket(*stream_info[:3])
     client.connect(stream_info[-1])
-
-    client.sendall(message.encode('utf8'))
+    if sys.version_info[0] == 2:
+        client.sendall(message)
+    else:
+        client.sendall(message.encode('utf8'))
 
     reply = ""
-    buffer_length = 10
+    buffer_length = 1024
     reply_complete = False
     while not reply_complete:
         part = client.recv(buffer_length)
         reply += (part.decode('utf8'))
         if len(part) < buffer_length:
             break
-    print(reply[:-3])
-    return reply[:-3]
+    print(reply)
+    return reply
     client.close()
```

Comment:
Why do you have `client.close()` after the `return` statement when you know that a `return` ends the function?